### ISSUE TITLE
Escaped dots in constraints don't validate against input names with dots

### ISSUE THREAD

**Issue by jtuds** [ansman/validate.js#188]
When an input name has a dot, e.g. `name="input.name"` and a constraint is added for that input, e.g. 
``` 
"input\\.name": {
    presence: true
}
```
the constraint will not validate against the name. This is because the escaping (of the dot) only happens when getting the input value and not when applying the errors to the inputs.

**Comment by ansman** [comment]
I'm a bit unclear on the problem, what are you expecting and what are you seeing?

**Comment by jtuds** [comment]
If I have an input field like this `<input type="text" name="foo.bar" />` and in the constraints object I have 
```
"foo.bar": {
    presence: true
}
```
Or
```
"foo\\.bar": {
    presence: true
}
```

I would expect the input field to be validated and to be checked for a presence of true, and then if presence is false, for an error to be added to the input. In the second constraints example I have escaped the period that appears in the input name (foo[period]bar). The input is validated, however the error is not correctly added to the input. I believe this is because escaping characters, i.e. "\\" are not removed from the input name before being ran through the validator. So in this instance, when applying the error, the validator is looking for for an input with the name `foo\\.bar` which doesn't exist. It should look for `foo.bar` without the slashes.

**Comment by roly445** [comment]
I am having the same issue here

**Comment by ansman** [comment]
Isn't the problem just that `collectFormValues` doesn't replace `.` with `\\.`?

**Comment by ansman** [comment]
A fix has been released in 0.12.0
